graph/src/util/log.rs

```diff
@@ -35,7 +37,10 @@ pub fn guarded_logger() -> (slog::Logger, slog_async::AsyncGuard) {
     (slog::Logger::root(drain.fuse(), o!()), guard)
 }
 
-pub fn register_panic_hook(panic_logger: slog::Logger) {
+pub fn register_panic_hook(
+    panic_logger: slog::Logger,
+    shutdown_sender: Mutex<Option<oneshot::Sender<()>>>,
```

Comment:
I think you can remove the `Mutex` and the `Option` around this, and then construct the mutex inside this function, to simplify the signature.

Reply:
Another thing, we should probably set the exit code to 1 inside the panic hook, since a panic means we'll terminate the process abnormally.

Reply:
I moved the sleep and exit into the panic_hook, so that now it sends the shutdown signal to main, sleeps, and then exits the process.

Reply:
The compiler didn't like constructing the `Mutex<Option<>>` inside of `register_panic_hook()`., so I am still passing the it in as an input.

Reply:
Could be because you did it inside `panic::set_hook`. But it's fine, this is not important.

Reply:
Yep, updated.